### PR TITLE
Feat/prod3 419 2

### PR DESCRIPTION
**Pull request by mendesmcmg**
# Description

- Adds monthly tickets job

## Does this pull request close any open issues?

- [#prod3-419](https://ribon.atlassian.net/browse/PROD3-419?atlOrigin=eyJpIjoiOTg4MTI2MzI0N2JmNGJlNDhhNDUyMTJmYjU2YWNiZTkiLCJwIjoiaiJ9)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

- call in your terminal
```
Tickets::GenerateClubMonthlyTicketsJob.perform_now(
      user:,
      platform: person_payment.subscription.platform,
      quantity: person_payment.subscription.offer.plan.monthly_tickets,
      integration: person_payment.integration
    )
   ```

### CODE REVIEW

File: app/controllers/users/v1/subscriptions_controller.rb [hunk 1]
```diff
@@ -19,13 +19,9 @@ def send_cancel_subscription_email
       end
 
       def member?
-        ids = current_user.customers.pluck(:id)
```
Comment (thaisribon) [inline app/controllers/users/v1/subscriptions_controller.rb:22]: here, I would suggest: 

          command = ::Users::VerifyClubMembership.new(user: current_user).call
          if command.success?
            is_member = command.result
            render json: { is_member: }, status: :ok
          else
            render_errors(command.errors)
          end

File: app/observers/person_payment_observer.rb [hunk 2]
```diff
@@ -21,4 +19,26 @@ def processing_to_failed?(person_payment)
...
+    Mailers::SendPersonPaymentEmailJob.perform_later(person_payment:)
+  end
+
+  def send_failed_payment_event(person_payment)
+    Events::PersonPayments::SendFailedPaymentEventJob.perform_later(person_payment:)
+  end
+
+  def give_monthly_tickets(person_payment)
+    user = person_payment.payer.user
+    return unless User::VerifyClubMembership(user).result
```
Comment (thaisribon) [inline app/observers/person_payment_observer.rb:35]: I think here is `Users::VerifyClubMembership.call(user:).result`

Comment (thaisribon) [reply app/observers/person_payment_observer.rb:35]: In fact, I think we can just leave this check at job

File: app/workers/tickets/generate_club_daily_tickets_worker.rb [hunk 3]
```diff
@@ -4,11 +4,11 @@ class GenerateClubDailyTicketsWorker
     sidekiq_options queue: :tickets
 
     def perform(*_args)
-      Subscription.active_from_club.each do |subscription|
-        Tickets::GenerateClubDailyTicketsJob.perform_later(user: subscription.payer.user,
-                                                           platform: subscription.platform,
-                                                           quantity: subscription.offer.plan.daily_tickets,
-                                                           integration: subscription.integration)
+      Subscription.where(status: :active).each do |subscriptions|
```
Comment (thaisribon) [inline app/workers/tickets/generate_club_daily_tickets_worker.rb:7]: I think it was a conflict problem, the red one is the correct one

File: app/observers/person_payment_observer.rb (hunk 2 above)
Comment (mendesmcmg) [reply app/observers/person_payment_observer.rb:35]: remove condition

File: app/observers/person_payment_observer.rb [hunk 4]
```diff
@@ -1,9 +1,17 @@
 class PersonPaymentObserver < ActiveRecord::Observer
   def after_update(person_payment)
-    Mailers::SendPersonPaymentEmailJob.perform_later(person_payment:) if processing_to_paid?(person_payment)
+    send_person_payment_email(person_payment) if processing_to_paid?(person_payment)
+    give_monthly_tickets(person_payment) if processing_to_paid?(person_payment)
```
Comment (thaisribon) [inline app/observers/person_payment_observer.rb:4]: this method proccessing_to_paid? is only for person_payments without subscriptions, we have to create another

**Review by thaisribon — APPROVED** [review]

**Pull request merged**